Added UAVOnix dialect from documentation.

queue-b opened the pull request:
NOTE: This does not match the published UAVonix.xml file provided by MAVLINK maintainers. That file is quite old.

Commit:
Added UAVOnix dialect from documentation.

NOTE: This does not match the published UAVonix.xml file
provided by MAVLINK maintainers.

Commit:
Added documentation and constants for Uavionix magic numbers.

queue-b commented:
@brandon-hass this is in support of ADS-B transponder support.  I added all the Uavionix messages to the UAVionix dialect.

Golint doesn't like the "self" parameter in the method definitions, but that matches the style of the  rest of the package.

brandon-hass reviewed (changes requested):
I feel like for the sake of consistency, creating the .xml file to generate the code and then modifying the CRC_EXTRA manually (unless you can get the field names to calculate it correctly, which would be ideal) would make more sense. Thoughts?

queue-b commented:
If you'd prefer I can definitely do that; assuming that works, you want the enum values in the XML as well?

queue-b commented:
Although -- if we have to manually modify the XMLCrc after generation, doesn't that break the guarantee of the generated code being correct for subsequent users?

queue-b commented:
Disregard last comment.  Looks like documentation lists byte offsets so we should be GTG.

Commit:
Added UAVIONIX xml file.

Commit:
Fixed a couple errors in the hand-generated uavionix dialect.

Commit:
Added XML-generated uavionix dialect.

NOTE: XMLCRCs were hand coded. Uavionix does not publish the XML
message name, meaning we can't just use the XML file and get good
 CRC values.

queue-b commented:
@brandon-hass Generated UAVionix dialect from the XML message set. CRCs needed to be manually changed because UAVionix doesn't actually give you the correct XML message name. I'm going to reach out and see if they will provide those.

queue-b commented:
@brandon-hass Also, it looks like the generator is producing incorrect output for the OWN_SHIP_STATIC message. The packing and unpacking functions seem to be incorrect. Any ideas/can you take a look?

Commit:
Replaced incorrectly generated OwnShipStatic message.

brandon-hass commented:
@botBoss It looks like uAvionix isn't ordering their fields correctly (or at least in the same way the other dialects do), which is causing the output that you're seeing. I thought it was an issue with the generator at first, but that doesn't seem to be the case.

brandon-hass commented on mavlink/definitions/uavionix.xml:
```diff
@@ -0,0 +1,399 @@
+<mavlink>
+    <include>common.xml</include>
+    <enums>
+      <enum name="UAVIONIX_SDA_LEVEL">
```
Since this is part of the uAvionix dialect, would there be a problem dropping the `UAVONIX_` prefix?

queue-b replied:
@brandon-hass I saw the other dialects doing prefixes like that, and the MAVLink common message set enums all start with MAVLink, so I thought that was the standard.

queue-b replied:
Well, OK, looking again at least the ualberta guys did :)

brandon-hass replied:
Ah, I see it in a few places. It just felt super verbose, but obviously doesn't matter.

queue-b replied:
I agree, it's unfortunate that there aren't any real standards in any of these definition files (or MAVLink as a whole).
